### PR TITLE
[ADAM-1697] Expand Illumina metadata regex to cover index sequences

### PR DESCRIPTION
**Pull request by pauldwolfe**
Now includes possible index sequence characters.

### CODE REVIEW

**Comment by pauldwolfe** [comment]
This change was discussed in gitter with @heuermh (actually he made the change, I just wrote the test and added the test resources). Perhaps he can review it.

File: adam-core/src/test/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDDSuite.scala [hunk 1]
```diff
@@ -576,4 +576,16 @@ class FragmentRDDSuite extends ADAMFunSuite {
 
     checkSave(variantContexts)
   }
+
+  sparkTest("Paired read names with index sequences in read names can grouped into fragments") {
+    val path1 = testFile("read_names_with_index_sequences_pair1.fq")
+    val path2 = testFile("read_names_with_index_sequences_pair1.fq")
```
Comment (heuermh) [inline adam-core/src/test/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDDSuite.scala]: I think this line should be `"read_names_with_index_sequences_pair2.fq"`

**Comment by heuermh** [comment]
Fixes #1697

**Comment by pauldwolfe** [comment]
Fixed change request from @heuermh

**Comment by heuermh** [comment]
Thank you, @pauldwolfe!

**Comment by heuermh** [comment]
Jenkins, test this please

**Comment by heuermh** [comment]
Jenkins, add to whitelist

**Comment by heuermh** [comment]
Squashed and merged as commit e7cbfcd, thank you, @pauldwolfe!

**Pull request merged**